### PR TITLE
some basic splitstore refactors

### PR DESCRIPTION
**Pull request by vyzo**
Preliminaries for implementing additional functionality on top (sortless compaction, gc, etc), this performs some basic refactoring to avoid polluting ballooning subsequent prs:
- The distinction between MarkSet and MarkSetVisitor is removed.
- SetConcurrent is removed from MarkSet interface; all mark sets must be concurrent, because ...
- Chain walking is now parallelized

### CODE REVIEW

File: blockstore/splitstore/splitstore_compact.go [hunk 1]
```diff
@@ -695,17 +703,42 @@ func (s *SplitStore) walkChain(ts *types.TipSet, inclState, inclMsgs abi.ChainEp
 		// the walk is BFS, so we can reset the walked set in every iteration and avoid building up
 		// a set that contains all blocks (1M epochs -> 5M blocks -> 200MB worth of memory and growing
 		// over time)
-		walked = cid.NewSet()
+		walked = newConcurrentVisitor()
 		walking := toWalk
 		toWalk = nil
```
Comment (magik6k) [inline blockstore/splitstore/splitstore_compact.go:708]: Technically we don't need this copy; `walking` could be set to nil after populating `workch`

**Review by magik6k — APPROVED** [review]
Don't see anything obviously wrong, just one tiny nitpick

File: blockstore/splitstore/splitstore_compact.go (hunk 1 above)
Comment (vyzo) [reply blockstore/splitstore/splitstore_compact.go:708]: ah yes, and we can reuse the slice; let me fix.

Comment (vyzo) [reply blockstore/splitstore/splitstore_compact.go:708]: hrm, we can't actually reuse the slice, it first comes from the tipset; will just setr to nil after copying.

Comment (vyzo) [reply blockstore/splitstore/splitstore_compact.go:708]: done.

Comment (vyzo) [reply blockstore/splitstore/splitstore_compact.go]: you know, we can first make a new one and copy when we initialize and then reuse; that will save somr allocations.
let me do.

Comment (vyzo) [reply blockstore/splitstore/splitstore_compact.go]: done done.

**Review by magik6k — APPROVED** [review]

**Pull request merged**